### PR TITLE
Remove odict compatibility layer since we now no longer support Python 2.6

### PR DESCRIPTION
**Pull request by astrofrog**
The slight issue with this one is that it was possible to import OrderedDict from ``astropy.utils.compat.odict`` as well as ``astropy.utils`` - in this PR, the deprecation warning is only emitted for the former. I've therefore made a fake OrderedDict class that subclasses the real one and raises a deprecation warning upon initialization.

### CODE REVIEW

**Comment by taldcroft** [comment]
I think the approach you took is fine.  My only comment is that it doesn't appear there is any test of the deprecation warning when importing the fake `OrderedDict` from `utils` or `utils.compat.odict`.

**Comment by astrofrog** [comment]
@taldcroft - I've added a test and removed a couple of further files that were no longer needed

**Comment by taldcroft** [comment]
@astrofrog - maybe you have done this already, but it might be worth locally stubbing the fake `OrderedDict` to raise an exception and then confirm that all other tests pass.  That way you know for sure that the local version is not getting used anywhere.

**Comment by astrofrog** [comment]
@taldcroft - yes, I did that and there are no more fixes.

File: astropy/utils/__init__.py [hunk 1]
```diff
@@ -17,4 +17,4 @@
 from .introspection import *
 from .misc import *
 
-from .compat.odict import OrderedDict
+from .compat.odict import OrderedDict
```
Comment (mhvk) [inline astropy/utils/__init__.py]: Missing end of line...

File: astropy/utils/tests/test_odict.py [hunk 2]
```diff
@@ -1,280 +1,17 @@
...
+    
+    with catch_warnings() as w:
+        OrderedDict()
+    
+    assert len(w) == 1
+    assert w[0].category == AstropyDeprecationWarning
+    assert w[0].message.args[0] == ("astropy.utils.compat.odict.OrderedDict is "
+                                    "now deprecated - import OrderedDict from "
+                                    "the collections module instead")
+
```
Comment (mhvk) [inline astropy/utils/tests/test_odict.py]: PEP8 - extra white space, no end of line

**Comment by mhvk** [comment]
Modulo two PEP8 issues, this looks good. Nice to see so much stuff disappear!

**Comment by astrofrog** [comment]
@mhvk - done!

**Comment by mhvk** [comment]
More out of curiosity: for `Fraction`, you give a deprecation warning upon importing anything, while here you give it when an `OrderedDict` imported from the old module is instantiated.  Warning upon import seems slightly more logical to me.

**Comment by astrofrog** [comment]
The difference is that for ``OrderedDict``, it was also imported into ``astropy.utils`` so the issue is that the deprecation warning would be seen when importing even just ``astropy.utils``, which would be annoying.

**Comment by mhvk** [comment]
OK, that makes sense. All looks OK, then!

**Comment by taldcroft** [comment]
Cruft be gone!

**Comment by astrofrog** [comment]
@taldcroft - since you like to get rid of stuff, feel free to review https://github.com/astropy/astropy/pull/4464, https://github.com/astropy/astropy/pull/4463, and https://github.com/astropy/astropy/pull/4462 :)

**Comment by pllim** [comment]
This is fun. It's like watching the intervention portion of The Hoarders.

Also, fun read: http://harrypotter.wikia.com/wiki/Vanishing_Spell

**Comment by astrofrog** [comment]
I forgot to remove the reference to the odict compat module in ``test_imports``, so I did it in master here: https://github.com/astropy/astropy/commit/b956583b477b15226caa57410de6ddfcaf0f8b9c

**Pull request merged**